sympy/solvers/solveset.py

```diff
@@ -120,11 +121,63 @@ def _invert_real(f, g_ys, symbol):
             return _invert_real(expo, imageset(Lambda(n, log(n)/log(base)),
                                                g_ys), symbol)
 
+    if isinstance(f, sin) or isinstance(f, csc):
+        n = Dummy('n')
+        if isinstance(g_ys, FiniteSet):
+            fi = sympify('a' + f.func.__name__)
+            # remove all the invalid elements
+            _a = []
+            for i in g_ys:
+                try:
+                    if i.is_real is S.false or \
+                        (isinstance(f, sin) and i not in Interval(-1, 1)) or \
+                        (isinstance(f, csc) and i in Interval(1, -1, True, True)):
+                            _a.append(i)
+                except TypeError:
```

Comment:
What causes these types error? In general it is bad practice ignore any error. Also, all the input validation should go into the `invert_real` function ideally this helper functions should only contain the logic.